test/jsonLongUnsigned.spec.js

```diff
@@ -0,0 +1,36 @@
+var expect = require('chai').expect;
+var Long = require('long');
+var JSONLong = require('../index');
+var JSONLongUnsigned = require('../index')({unsigned: true});
+var JSONLongSigned = require('../index')({unsigned: false});
+
+describe("Tests the Long support with unsigned long values", function(){
+
+    it("tests the JSON parse and stringify functionality with unsigned long values", function(done){
+        var jsonString = '{"longValue":18446744073709551614}';
```

Comment:
it seems odd to me that when we get the stringfied JSON output, there isn't anything there to indicate whether the value is signed or not

imo, that's a problem, no?

Reply:
I don't think it is a problem. We are just using `toString()` method of the Long instance.

What we have is just a number, it's value is not related to it's bit representation from JSON point of view. 
Also this JSON string might have been created by a different JSON library that probably doesn't care.

What did you have in mind for an indication?

Reply:
If I understand correctly; we should fail when we have `{unsigned: true}` and we want to stringify a Long object that is signed. Same for the reverse, i.e.:

```js
var signedLong = new Long("42", false);
var unsignedLong = new Long("42", true);

JSONUnsignedLong.stringify(unsignedLong);  // OK
JSONUnsignedLong.stringify(signedLong);    // Fail
JSONSignedLong.stringify(unsignedLong);    // Fail
JSONSignedLong.stringify(signedLong);      // OK
```

But I think we should do this if the `unsigned` option is present, not by default. Otherwise people cannot mix signed and unsigned Long in an object. Alternatively, we can allow mixing as an option as well. So:
```js
JSONLong.stringify(signedLong);    // OK
JSONLong.stringify(unsignedLong);  // OK
```

Reply:
Maybe I am missing something - can the "unsigned" flag be true, false and undefined?

If not, it must have a default value, whether it's ``true`` or ``false``. And, if that's the case, I don't understand your suggestion.

If yes, three possible settings exist, then I am ok with what you are suggesting.

But in either case, we need to bake this in with a test.